modules/tinymce/src/core/main/ts/caret/CaretContainerInput.ts

```diff
@@ -23,6 +23,7 @@ const findBlockCaretContainer = (editor: Editor): HTMLElement | null =>
 const showBlockCaretContainer = (editor: Editor, blockCaretContainer: HTMLElement): void => {
   if (blockCaretContainer.hasAttribute('data-mce-caret')) {
     CaretContainer.showCaretContainerBlock(blockCaretContainer);
+    editor.selection.setRng(editor.selection.getRng()); // Clears the fake caret state
```

Comment:
This was a regression when the IE specific things got removed. It was doing more than just IE control selection removal. I added a test for this as well since we didn't have one that failed.